### ISSUE TITLE
Performance Regression?

### ISSUE THREAD

**Issue by BrunoJuchli** [Fody/Fody#547]
I suspect there was a significant performance regression introduced with one of the latest Fody, PropertyChanged, Virtuosity, EmptyConstructor or Equals releases.
I'm now seeing output like:

> 2>    Fody: Fody (version 3.2.0.0) Executing
2>    Fody:   Finished Fody 4852ms.
2>    Fody:   Skipped Verifying assembly since it is disabled in configuration

while before usually i've seen times in the lower 3 digits exclusively.

Has anyone made similar observations?

@SimonCropp 
I don't expect you to do anything; I'll investigate further as to how the performance was in previous versions.

**Comment by SimonCropp** [comment]
yeah not sure what could have cause that. suggest an elimination test. gradually remove weavers till u find the culprit, or isolate it to fody itself.

**Comment by BrunoJuchli** [comment]
I've compared a current and an older build:

Fody 3.0.4 vs 3.2.4
PropertyChanged.Fody 2.5.7 vs 2.5.11
EmptyConstructor 1.4.1 vs 1.4.2
Virtuosity 1.21.0 vs 1.21.0

And can't say that the new versions are slower. It's more that *sometimes* the time varies quite significantly between runs (observed up to a factor of 10). It's not that a particular solution-build is particularly slow, it's more that weaving the assemblies varies significantly. Given that's non-realtime code it's not unexpected, though. Performance overall seems stable (making an accurate statement from based on the few builds I've looked at is not possible...).

**Comment by SimonCropp** [comment]
perhaps it is VS recycling or switching msbuild.exe processes?

**Comment by SimonCropp** [comment]
@BrunoJuchli i think your problem might be due to this https://github.com/Fody/PropertyChanged/issues/363#issuecomment-424721815

does the perf improve if u do a build after purging your nuget cache?

**Comment by BrunoJuchli** [comment]
I'll check.
We're using using classic style csproj with packages.config though.

**Comment by BrunoJuchli** [comment]
WPF + new csproj work still don't work well together, right? AFAIR the plan is to fix this for Visual Studio vNext...
We're planning to move to new package references, though. That's supported with the old csproj. It just didn't get enough priority yet, though ;-)

**Comment by BrunoJuchli** [comment]
> I have been informed by both @distantcam and @GeertvanHorrik that wpf works in the new project system

That's interesting. Last time I tried a few months ago you still needed quite some workarounds and even then it didn't work flawlessly:
https://github.com/aienabled/WpfNewProjectSystemSample

My memory tells me that I read that we could not expect this to get fixed until WPF support in .Net Core (which should come end of 2018 - beginning of 2019).

The issue is not closed either: https://github.com/dotnet/project-system/issues/1467

So I'm wondering...?

**Comment by GeertvanHorrik** [comment]
@BrunoJuchli Please check out https://github.com/WildGums/LogViewer/blob/develop/src/LogViewer/LogViewer.csproj. Not sure what you mean by workarounds, but this works fine for all our WPF apps and components (we migrated *everything* to the new project structure).

If you feel this is a workaround, that's fine. But I think it's important to understand that the time of the Fody developers is limited and they carefully need to review where they are putting their effort. I think it's reasonable for the Fody developers to focus on where the market is (going), and that's the new project structure. There are working solutions, and after migrating I can only recommend how easy it was and is to work with the new project structure :-)

**Comment by tom-englert** [comment]
@GeertvanHorrik in industrial dimensions "where the market is going" means it will arrive there earliest in 10 years. 
When the first men landed on the moon, they used technology that was at least 10 years old due to the project lifetime. 
And they probably would have failed if they had continuously migrated their technology like early adopters.

**Comment by GeertvanHorrik** [comment]
> And they probably would have failed if they had continuously migrated their technology like early adopters.

Definitely, that's why the project migration wasn't done when .NET Core 1.0 was released, but it's already matured over the years.
